yqlplus_engine/src/test/java/com/yahoo/yqlplus/engine/sources/SampleResultSource.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2019 Yahoo Inc.
+ * Licensed under the terms of the Apache version 2.0 license.
+ * See LICENSE file for terms.
+ */
+
+package com.yahoo.yqlplus.engine.sources;
+
+import java.util.Arrays;
+import java.util.List;
+
+import com.yahoo.yqlplus.api.Source;
+import com.yahoo.yqlplus.api.annotations.Query;
+
+public class SampleResultSource implements Source {
+    
+    private Sample field1 = new Sample();
+    
+    private Sample field2 = new Result<Sample>(field1).getResult();
```

Comment:
Is this the test case you mentioned? For this case no cast needed.

Reply:
I mean a case where there is no access method and the field is public -- so it uses the field-reading code rather than the method-calling code.

Reply:
For public field got java.lang.NoSuchFieldError.

Reply:
Fixed that case as well. The root cause is should use returnType not genericReturnType.